### PR TITLE
input checks and unit tests for TwoColumnOperatorTransformer

### PR DESCRIPTION
**Pull request by ChaitanMohr**
This PR aims to edit the existing TwoColumnOperatorTransformer to ensure that:
- a value error is raised when instantiated with empty columns list
- a type error is raised when columns instantiated with a string

### CODE REVIEW

File: tests/numeric/test_TwoColumnOperatorTransformer.py [hunk 1]
```diff
@@ -19,6 +19,27 @@ def example_transformer():
...
+                pd_method_kwargs={"axis": 1},
+            )
+
+    def test_column_size_error(self):
+        """Checks that the column is of length 2."""
+        with pytest.raises(
+            ValueError,
+            match="columns must be a list containing two column names",
+        ):
+            TwoColumnOperatorTransformer("mul", [], "c", pd_method_kwargs={"axis": 1})
```
Comment (TommyMatthews) [inline tests/numeric/test_TwoColumnOperatorTransformer.py:41]: Please can you check this with a list of length one? 

An empty list will evaluate to none and so could accidentally be picking up some other behaviour (I'm maybe being over cautious here)

File: CHANGELOG.rst [hunk 2]
```diff
@@ -34,6 +34,12 @@ Removed
 this functionality. This required updating a lot of test files.
 - The `columns_set_or_check()` method from BaseTransformer. With the above change it was no longer necessary. Subsequent updates to nominal transformers and their tests were required.
 
+1.2.3 (2024-02-23)
```
Comment (TommyMatthews) [inline CHANGELOG.rst:37]: Changelog is out of date, please can you pull the latest changes from develop to your branch :)

File: tubular/_version.py [hunk 3]
```diff
@@ -1 +1 @@
-__version__ = "1.3.0"
+__version__ = "1.2.3"
```
Comment (TommyMatthews) [inline tubular/_version.py:1]: Same as changelog - we're on 1.3.0 now!

File: tests/numeric/test_TwoColumnOperatorTransformer.py (hunk 1 above)
Comment (ChaitanMohr) [reply tests/numeric/test_TwoColumnOperatorTransformer.py:41]: all good with both empty and length one it seems

**Comment by davidhopkinson26** [comment]
Hi @ChaitanMohr, please add a PR description.  I know we have the linked issue but a descriptive PR description makes it clear what occured in this PR for future contributors.

File: CHANGELOG.rst (hunk 2 above)
Comment (davidhopkinson26) [reply CHANGELOG.rst:37]: main is on 1.3.0 now: 
 https://github.com/lvgig/tubular/blob/main/CHANGELOG.rst

Comment (limlam96) [reply CHANGELOG.rst:37]: Can you move this entry to under 1.3.0 please @ChaitanMohr ?

Comment (ChaitanMohr) [reply CHANGELOG.rst:37]: https://github.com/lvgig/tubular/pull/183/commits/d1d8044bf28cb5a0cc02eef9ce2f50ecad719cdf

File: tests/numeric/test_TwoColumnOperatorTransformer.py (hunk 1 above)
Comment (ChaitanMohr) [reply tests/numeric/test_TwoColumnOperatorTransformer.py]: included both tests for empty and length 1 https://github.com/lvgig/tubular/pull/183/commits/a25f43968a110293324dbaa518dec1290a6b2117

File: CHANGELOG.rst (hunk 2 above)
Comment (limlam96) [reply CHANGELOG.rst:37]: Hm looks like the changelog is unchanged?

Comment (ChaitanMohr) [reply CHANGELOG.rst]: updated changelog https://github.com/lvgig/tubular/pull/183/commits/2544a1f132029d839c7c2374c2a40e93c86c6da5

**Review by limlam96 — APPROVED** [review]

**Pull request merged**